File: api/avatar.js
const crypto = require("crypto");

module.exports = (req, res) => {
  const { uid } = req.query;

  if (!uid) {
    return res.status(400).send("UID parameter required");
  }

  const totalAvatars = 15;

  const hash = crypto.createHash("md5").update(uid).digest("hex");
  const number = parseInt(hash, 16);

  const avatarNumber = (number % totalAvatars) + 1;

  res.redirect(`/avatars/${avatarNumber}.png`);
};

File: api/avatar.py
import hashlib

def handler(request):
    uid = request.args.get("uid")

    if not uid:
        return {
            "statusCode": 400,
            "body": "UID parameter required"
        }

    total_avatars = 15

    hash_value = int(hashlib.md5(uid.encode()).hexdigest(), 16)
    avatar_number = (hash_value % total_avatars) + 1

    return {
        "statusCode": 302,
        "headers": {
            "Location": f"/avatars/{avatar_number}.png"
        }
    }
